app/models/services/base-service.ts

```diff
@@ -88,6 +88,23 @@ const conditionToOperationField = {
   temporalSubset: 'shouldTemporalSubset',
 };
 
+/**
+ * Step operations that are aggregating steps
+ */
+const aggregatingOperations = [
+  'concatenate',
+];
+
+/**
+ *  Returns true if the workflow step aggregates output from the previous step
+ * (and therefore must wait for all output before executing)
+ * @param step - the step in a workflow
+ * @returns true if the step is an aggregating step, false otherwise
+ */
+function _stepHasAggregatedOutput(step: ServiceStep): boolean {
```

Comment:
Nit: I thought our convention was to only prefix with _ on methods within a class. At the top level our indication a function was private was whether or not it was exported.

Reply:
Oh - for some reason I thought it was anything that wasn't exported. I'll change it.

Reply:
Done.

Reply:
I want to say this convention was your (previous James') idea, and we all liked it better.